Windows buildpack and .NET Core (Linux) (#82)
DataDog/datadog-cloudfoundry-buildpack: README.md

#### General configuration of the Datadoc Agent
All the options supported by the Agent in the main configuration file (`lib/dist/datadog.yaml`) can also be set through environment variables as described in the [documentation of the Agent](https://github.com/DataDog/datadog-agent/blob/master/docs/agent/config.md#environment-variables).

> #### .NET Traces

Reviewer: 🙇 👏

Author: Can't take the credit, mostly comes from the great work @zippolyte did - https://github.com/DataDog/datadog-cloudfoundry-buildpack/pull/61/files#diff-04c6e90faac2675aa89e2176d2eec7d8R82

Reviewer 2: 🙇

> The buildpack also includes the [.NET Tracer](https://docs.datadoghq.com/tracing/setup/dotnet/?tab=netframeworkonwindows) for the .NET Framework on windows cells. To start instrumenting your app, perform the following:

Reviewer 3: ```suggestion
The buildpack also includes the [.NET Tracer](https://docs.datadoghq.com/tracing/setup/dotnet/?tab=netframeworkonwindows) for the .NET Framework on windows cells. To start instrumenting your app:
```

**example**:
```xml
<runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
        <probing privatePath="C:\Users\vcap\app\datadog\dotNetTracer" />
    </assemblyBinding>
</runtime>
```

Alternatively, you can install the [Datadog.Trace.ClrProfiler.Managed Nuget package](https://www.nuget.org/packages/Datadog.Trace.ClrProfiler.Managed) in your app before pushing it to CloudFoundry.

2. Set the following environment variables in your application:
```
# Set your API key to send traces to Datadog
cf set-env $YOUR_APP_NAME DD_API_KEY $YOUR_DATADOG_API_KEY
# Setup the dotnet datadog tracer
cf set-env $YOUR_APP_NAME COR_ENABLE_PROFILING 1
cf set-env $YOUR_APP_NAME COR_PROFILER {846F5F1C-F9AE-4B07-969E-05C26BC060D8}
cf set-env $YOUR_APP_NAME COR_PROFILER_PATH C:\Users\vcap\app\datadog\dotNetTracer\Datadog.Trace.ClrProfiler.Native.dll
cf set-env $YOUR_APP_NAME DD_TRACE_LOG_PATH "\Users\vcap\app\datadog\dotNetTracer\tracer.log"
cf set-env $YOUR_APP_NAME DD_INTEGRATIONS "\Users\vcap\app\datadog\dotNetTracer\integrations.json"
cf set-env $YOUR_APP_NAME DD_TRACE_ENABLED true
# Restage your app to take these changes
cf restage $YOUR_APP_NAME
```

### DogStatsD Away!

> You're all set up to use DogStatsD. Import the relevant library and start sending data! To learn more, [check our our documentation](https://docs.datadoghq.com/guides/DogStatsD/). Additionally, we have [a list of DogStatsD libraries](https://docs.datadoghq.com/libraries/) you can check out to find one that's compatible with your application.

Reviewer 3: ```suggestion
You're all set up to use DogStatsD. Import the relevant library and start sending data! To learn more, [check our our documentation](https://docs.datadoghq.com/guides/DogStatsD/). Additionally, there is [a list of DogStatsD libraries](https://docs.datadoghq.com/libraries/) you can check out to find one that's compatible with your application.
```

**example**:
```xml
<runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
        <probing privatePath="C:\Users\vcap\app\datadog\dotNetTracer" />
    </assemblyBinding>
</runtime>
```

2. Setup the appropriate environment variables for your application depending on the OS the app will run on. These can be set directly in the `run.cmd` file.

**example**:
```
# Setup the Tracing Environment Variables
SET COR_ENABLE_PROFILING=1
SET COR_PROFILER={846F5F1C-F9AE-4B07-969E-05C26BC060D8}
SET COR_PROFILER_PATH_64=C:\Users\vcap\app\datadog\dotNetTracer\x64\Datadog.Trace.ClrProfiler.Native.dll
SET COR_PROFILER_PATH_32=C:\Users\vcap\app\datadog\dotNetTracer\x86\Datadog.Trace.ClrProfiler.Native.dll
SET DD_INTEGRATIONS=\Users\vcap\app\datadog\dotNetTracer\integrations.json
SET DD_DOTNET_TRACER_HOME=\Users\vcap\app\datadog\dotNetTracer
SET DD_TRACE_LOG_PATH=\Users\vcap\app\datadog\dotNetTracer\dotnet-profiler.log
# Redirect logs to the Datadog Agent
.cloudfoundry\hwc.exe | powershell C:\Users\vcap\app\datadog\scripts\redirect_logs.ps1 2>&1
```

> Alternatively, you can install the [Datadog.Trace.ClrProfiler.Managed Nuget package](https://www.nuget.org/packages/Datadog.Trace.ClrProfiler.Managed) in your app before pushing it to CloudFoundry.

Reviewer 4: ```suggestion
Alternatively, you can install the [Datadog.Trace.ClrProfiler.Managed Nuget package](https://www.nuget.org/packages/Datadog.Trace.ClrProfiler.Managed) in your application before pushing it to CloudFoundry.
```

### Linux (.NET Core Apps)

Set the following environment variables in your application:
```
# Set your API key to send traces to Datadog
cf set-env $YOUR_APP_NAME DD_API_KEY $YOUR_DATADOG_API_KEY
# Setting this environment variable will bundle the .NET Tracing dependencies and enable the
cf set-env $YOUR_APP_NAME DD_DOTNET_TRACING true
# environment variables needed to automatically instrument your .NET Application
cf restage $YOUR_APP_NAME
```

> ### DogStatsD Away!

Reviewer 4: ```suggestion
### DogStatsD Away!

```

> You're all set up to use DogStatsD. Import the relevant library and start sending data! To learn more, [check our our documentation](https://docs.datadoghq.com/guides/DogStatsD/). Additionally, there is [a list of DogStatsD libraries](https://docs.datadoghq.com/libraries/) you can check out to find one that's compatible with your application.

Reviewer 4: What are the lib you are referring to here? Dogstatsd libs? 

```suggestion
You're all set up to use DogStatsD. Import the relevant library and start sending data! To learn more, [check the Datadog DogStastD documentation](https://docs.datadoghq.com/guides/DogStatsD/). Additionally, there is [a list of DogStatsD libraries](https://docs.datadoghq.com/libraries/) you can check out to find one that's compatible with your application.
```

Author: Yeah, like the C# one if you have a .NET app, the ruby one for ruby apps, etc.

> Note: On Windows .NET applications you can add the Dogstatsd-CSharp client to your project's dependencies to submit custom metrics from your application. This requires publishing the application (to include the dogstatsd client dependency) and passing the published folder in your app manifest.

Reviewer 4: ```suggestion
**Note**: On Windows .NET applications you can add the [Dogstatsd-CSharp client](https://github.com/DataDog/dogstatsd-csharp-client) to your project's dependencies to submit custom metrics from your application. This requires publishing the application (to include the DogStatsD client dependency) and passing the published folder in your application manifest.
```
